vita3k/codec/src/mjpeg.cpp

```diff
@@ -262,75 +347,76 @@ bool MjpegDecoderState::receive(uint8_t *data, DecoderSize *size) {
         return false;
     }
 
+    if (mcu_width == 0 || mcu_height == 0) {
+        LOG_WARN("Mjpeg MCU size is not set. Falling back to 1x1.");
+        mcu_width = mcu_height = 1;
+    }
+
+    // Calculate MCU-aligned dimensions
+    uint32_t aligned_width = align_dimension(frame->width, mcu_width);
+    uint32_t aligned_height = align_dimension(frame->height, mcu_height);
+
+    if (frame->width != aligned_width || frame->height != aligned_height) {
+        LOG_INFO("Mjpeg frame dimensions are not MCU-aligned. MCU size: {}x{}, frame size: {}x{} -> aligned size: {}x{}.",
+            mcu_width, mcu_height, frame->width, frame->height, aligned_width, aligned_height);
+    }
+
     if (data) {
+        ChannelInfo channel_info[3];
+        uint32_t total_size = aligned_width * aligned_height;
```

Comment:
```suggestion
        uint32_t first_channel_size = aligned_width * aligned_height;
```
The name `total_size` is a bit misleading, that's not the total size of the image, but only of the first channel/plane (feel free to change the name to something similar with the same intent)